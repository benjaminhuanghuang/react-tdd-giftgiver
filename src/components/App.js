import React, { Component } from 'react'
import { Button } from 'react-bootstrap'

export default class App extends Component {
  constructor() {
    super();
    this.state = {
      gifts: []
    };
  }
  addGift = () => {
    const { gifts } = this.state;

    const ids = this.state.gifts.map(gift => gift.id)
    const max_id = ids.legth > 0 ? Math.max(...ids) : 0
    gifts.push({ id: max_id + 1 })

    this.setState({gifts})
  }
  render() {
    return (
      <div>

        <Button className='btn-add' onClick={this.addGift}> Add Gift</Button>
      </div>

    )
  }
}
