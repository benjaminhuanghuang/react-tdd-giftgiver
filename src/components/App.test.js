import React from "react"
import { shallow } from 'enzyme'
//
import App from './App'

const app = shallow(<App />)

it("renders correnctly", () => {
  expect(app).toMatchSnapshot()
})


it("renders the state with an empyt list of gift", () => {
  expect(app.state().gifts).toEqual([])
})

it("add a new gift to state when click the add gift button", () => {
  app.find('.btn-add').simulate('click');
  expect(app.state().gifts).toEqual([{ id: 1 }])
})