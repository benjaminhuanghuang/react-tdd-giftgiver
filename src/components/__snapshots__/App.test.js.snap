// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correnctly 1`] = `ShallowWrapper {}`;

exports[`renders the state with an empyt list of gift 1`] = `ShallowWrapper {}`;
